components/org.wso2.carbon.multitenancy.tenant.service/src/main/java/org/wso2/carbon/multitenancy/tenant/service/kubernetes/KubernetesTenancyProvider.java

```diff
@@ -31,16 +34,31 @@
 /**
  * Tenant provider for Kubernetes cluster manager.
  */
-public class KubernetesTenancyProvider extends KubernetesBase implements TenancyProvider {
+public class KubernetesTenancyProvider implements TenancyProvider {
 
+    private static final Logger logger = LoggerFactory.getLogger(KubernetesTenancyProvider.class);
+
+    private static final String KUBERNETES_MASTER_ENV_VAR_NAME = "KUBERNETES_MASTER";
+    private static final String KUBERNETES_MASTER_SYS_PROPERTY_NAME = "kubernetes.master";
+    private static final String ARTIFACTS_PATH_ENV_VAR_NAME = "WSO2_KUBERNETES_ARTIFACTS_PATH";
+    private static final String ARTIFACTS_PATH_SYS_PROPERTY_NAME = "wso2.kubernetes.artifacts.path";
     private static final String RESERVED_NAMESPACE_DEFAULT = "default";
     private static final String RESERVED_NAMESPACE_KUBE_SYSTEM = "kube-system";
 
+    private final DefaultKubernetesClient kubernetesClient;
+
     /**
-     * Initializes the Kubernetes client by providing the master node endpoint.
+     * Initializes the Kubernetes kubernetesClient by providing the master node endpoint.
      */
     public KubernetesTenancyProvider() {
-        super();
+        String kubernetesMasterUrl = System.getenv(KUBERNETES_MASTER_ENV_VAR_NAME);
+        if (kubernetesMasterUrl == null || kubernetesMasterUrl.isEmpty()) {
+            kubernetesMasterUrl = System.getProperty(KUBERNETES_MASTER_SYS_PROPERTY_NAME);
+        }
+        kubernetesClient = new DefaultKubernetesClient(kubernetesMasterUrl);
```

Comment:
A very good catch Thusitha! I just fixed it and pushed changes. Please review. Thanks!